crates/sui-indexer/src/balance_changes.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+use std::ops::Neg;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use move_core_types::language_storage::TypeTag;
+use tokio::sync::RwLock;
+
+use sui_json_rpc_types::BalanceChange;
+use sui_json_rpc_types::{SuiExecutionStatus, SuiTransactionEffects, SuiTransactionEffectsAPI};
+use sui_types::base_types::{MoveObjectType, ObjectID, SequenceNumber};
+use sui_types::coin::Coin;
+use sui_types::gas::GasCostSummary;
+use sui_types::gas_coin::GAS;
+use sui_types::object::{Object, Owner};
+
+use crate::errors::IndexerError;
+use crate::store::PgIndexerStore;
+
+pub async fn get_balance_changes_from_effect<P: ObjectProvider<Error = E>, E>(
+    object_provider: &P,
+    effects: &SuiTransactionEffects,
```

Comment:
a couple of reasons to put these in sui-indexre:
- the trait is defined in json rpc, which is a dependency crate indexer
- secondary one: json rpc types like SuiTransactionEffects are used in indexer

Reply:
Is this a rust thing where traits need to copied? Like we can't just import the `ObjectProvider` trait

Reply:
doing it wrong, I should do that indexer indeed, but not for arc